tests/unit/options.js

```diff
@@ -2932,3 +2932,27 @@ exports.module.declarationRestrictions = function( test ) {
 
   test.done();
 };
+
+exports["gh-2436"] = function( test ) {
+  // see gh-2436. Previously jshint directives after a function declaration were ignored.
+  TestRun(test)
+    .addError(12, "This function has too many parameters. (2)")
+    .test([
+      "function x() {}",
+      "/*jshint evil: true */", // <-- previously ignored
+      "eval();",
+      "// jshint maxparams: 2",
+      "function Foo () {}",
+      "// jshint maxparams: 9", // <-- previously ignored
+      "Foo.prototype.bar = function (a, b, c, d, e) {",
+      "  return a + b + c + d + e;",
+      "};",
+      "var z = () => foo;",
+      "// jshint maxparams: 1",
+      "function y(a, b) {}",
+      "// jshint evil: false",
+      "eval()/*jshint evil: true */;"
```

Comment:
yes it is. It is kind of a breaking change, so it might need a major release, upto @jugglinmike. Its hard to fix the bug properly and not introduce a breaking change. Its another one of those where the existing behaviour is probably not expected so we have to guess whether this will likely effect many people.
There is another reason why it might be worth waiting to major, which I'll comment in a sec.